tests/components/vizio/test_init.py

```diff
@@ -0,0 +1,42 @@
+"""Tests for Vizio init."""
+import pytest
+
+from homeassistant.components.media_player.const import DOMAIN as MP_DOMAIN
+from homeassistant.components.vizio.const import DOMAIN
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.setup import async_setup_component
+
+from .const import MOCK_USER_VALID_TV_CONFIG, UNIQUE_ID
+
+from tests.common import MockConfigEntry
+
+
+async def test_setup_component(
+    hass: HomeAssistantType,
+    vizio_connect: pytest.fixture,
+    vizio_update: pytest.fixture,
+) -> None:
+    """Test component setup."""
+    assert await async_setup_component(
+        hass, DOMAIN, {DOMAIN: MOCK_USER_VALID_TV_CONFIG}
+    )
+    await hass.async_block_till_done()
+    assert len(hass.states.async_entity_ids(MP_DOMAIN)) == 1
+
+
+async def test_load_and_unload(
+    hass: HomeAssistantType,
+    vizio_connect: pytest.fixture,
+    vizio_update: pytest.fixture,
+) -> None:
+    """Test loading and unloading entry."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN, data=MOCK_USER_VALID_TV_CONFIG, unique_id=UNIQUE_ID
+    )
+    await hass.config_entries.async_add(config_entry)
```

Comment:
Don't call `async_add`. Just set the entry up with `config_entries.async_setup`.

Reply:
Doesn't the entry first have to be added before it is setup? Are you suggesting I use entry.add_to_hass() + async_setup() instead?

Reply:
I did this ^